src/server/command/worldedit/blueprint.zig

```diff
@@ -0,0 +1,193 @@
+		}
+		argsList.deinit();
+	}
+
+	if(argsList.items.len < 1) {
+		source.sendMessage("#ff0000Not enough arguments for /blueprint, expected at least 1.", .{});
+		return;
+	}
+	const subcommand = BlueprintSubCommand.fromString(argsList.items[0]);
+	_ = switch(subcommand) {
+		.save => blueprintSave(argsList, source),
+		.delete => blueprintDelete(argsList, source),
+		.load => blueprintLoad(argsList, source),
+		.list => blueprintList(argsList, source),
+		.other => {
+			source.sendMessage("#ff0000Unrecognized subcommand for /blueprint: '{s}'", .{argsList.items[0]});
+		},
+		.empty => {
+			source.sendMessage("#ff0000Missing subcommand for **/blueprint**, usage: {s} ", .{usage});
+		},
+	} catch |err| {
+		std.log.info("Error: {s}", .{@errorName(err)});
+		source.sendMessage("#ff0000Error: {s}", .{@errorName(err)});
+	};
+}
+
+fn blueprintSave(args: List([]const u8), source: *User) !void {
+	if(args.items.len < 2) {
+		source.sendMessage("#ff0000**/blueprint save** requires FILENAME argument.", .{});
+		return;
+	}
+	if(args.items.len >= 3) {
+		source.sendMessage("#ff0000Too many arguments for **/blueprint save**. Expected 1 argument, FILENAME.", .{});
+		return;
+	}
+	source.mutex.lock();
+	defer source.mutex.unlock();
+
+	if(source.commandData.clipboard) |clipboard| {
+		const storedBlueprint = clipboard.store(main.stackAllocator);
+		defer main.stackAllocator.free(storedBlueprint);
+
+		const fileName: []const u8 = ensureBlueprintExtension(main.stackAllocator, args.items[1]);
+		defer main.stackAllocator.free(fileName);
+
+		var cwd = std.fs.cwd();
+
+		_ = cwd.makeDir("blueprints") catch null;
+
+		var blueprintsDir = try cwd.openDir("blueprints", .{});
+		defer blueprintsDir.close();
+
+		std.log.info("Saving clipboard to blueprint file: {s}", .{fileName});
+		source.sendMessage("#00ff00Saving clipboard to blueprint file: {s}", .{fileName});
+
+		try blueprintsDir.writeFile(.{
+			.sub_path = fileName,
+			.data = storedBlueprint,
+			.flags = .{.lock = .exclusive},
+		});
+	} else {
+		source.sendMessage("#ff0000Error: No clipboard content to save.", .{});
+	}
+}
+
+fn ensureBlueprintExtension(allocator: NeverFailingAllocator, fileName: []const u8) []const u8 {
+	if(!std.ascii.endsWithIgnoreCase(fileName, ".blp")) {
+		return std.fmt.allocPrint(allocator.allocator, "{s}.blp", .{fileName}) catch unreachable;
+	} else {
+		return allocator.dupe(u8, fileName);
+	}
+}
+
+fn blueprintDelete(args: List([]const u8), source: *User) !void {
+	if(args.items.len < 2) {
+		source.sendMessage("#ff0000**/blueprint delete** requires FILENAME argument.", .{});
+		return;
+	}
+	if(args.items.len >= 3) {
+		source.sendMessage("#ff0000Too many arguments for **/blueprint delete**. Expected 1 argument, FILENAME.", .{});
+		return;
+	}
+
+	const fileName: []const u8 = ensureBlueprintExtension(main.stackAllocator, args.items[1]);
+	defer main.stackAllocator.free(fileName);
+
+	var cwd = std.fs.cwd();
+
+	_ = cwd.makeDir("blueprints") catch null;
```

Comment:
Should use `try` as well, like the command below it.

(Also instead `_ = ... catch null` you could do `... catch {}` to return a void type.)

Reply:
Doesn't it fail when path already exist?

Reply:
Oh right it does. Either way you should use main.files
(Or at least openPath (which creates the directory automatically) instead of openDir)